File: iosApp/iosApp/Presentation/Main/CafeManagement/CastManagement/CastManagementUiState.swift
//
//  CastManagementUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation

enum CastScheduleViewMode: String, CaseIterable {
    case week = "cast_management_view_week"
    case month = "cast_management_view_month"
}

struct CastManagementUiState {
    var isLoading: Bool = true
    var cafeName: String = ""
    var viewMode: CastScheduleViewMode = .week
    var periodStart: Date? = nil
    var periodEnd: Date? = nil
    var weekColumns: [WeekColumn] = []
    var monthOffset: Int = 0
    var monthCells: [MonthCell] = []
    var errorMessage: String?

    struct WeekColumn: Identifiable {
        let id: String
        let dayLabel: String
        let dateLabel: String
        let castNames: [String]
    }

    struct MonthCell: Identifiable {
        let id: String
        let dayNumber: Int
        let castNames: [String]
    }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/castmanagement/CastManagementUiState.kt
package com.hhp227.concafe.presentation.main.cafemanagement.castmanagement

enum class CastScheduleViewMode { WEEK, MONTH }

data class CastManagementUiState(
    val isLoading: Boolean = true,
    val cafeName: String = "",
    val viewMode: CastScheduleViewMode = CastScheduleViewMode.WEEK,
    val periodStart: String = "",
    val periodEnd: String = "",
    val weekColumns: List<WeekColumn> = emptyList(),
    val monthOffset: Int = 0,
    val monthCells: List<MonthCell> = emptyList(),
    val errorMessage: String? = null
) {
    data class WeekColumn(
        val dayLabelKey: String,
        val dateLabel: String,
        val castNames: List<String>
    )

    data class MonthCell(
        val dayNumber: Int,
        val castNames: List<String>
    )
}
